src/refinement/ptq.jl

```diff
@@ -0,0 +1,61 @@
+# ------------------------------------------------------------------
+# Licensed under the MIT License. See LICENSE in the project root.
+# ------------------------------------------------------------------
+
+"""
+    PTQ()
+
+Refinement of triangle meshes.
+A triangle is subdivided into four triangles.
+"""
+struct PTQ <: RefinementMethod end
+
+function refine(mesh, ::PTQ)
+  primal_triangle_quadrisection(mesh, (p1, p2) -> Point((coordinates(p1) + coordinates(p2))/2))
```

Comment:
Are there plans to support other types of "middle" function?  I think this function primal_triangle_quadrisection could be the refine function. No need to introduce a separate function.

Reply:
Yes, the hyperbolic midpoint to construct hyperbolic triangles. Also if you take three points on the unit sphere and you iterate the PTQ with the normalized midpoint, you get a spherical triangle.

Reply:
We can refactor our refinement methods later to add this flexibility and avoid code duplication. As far as I remember our CatmullClark method is the same as TriRefinement with a custom "middle" function as well.